pkg/platform/local/platform.go

```diff
@@ -693,18 +633,12 @@ func (p *Platform) getFunctionHTTPPort(createFunctionOptions *platform.CreateFun
 
 	// if there was a previous deployment and no configuration - use that
 	if previousHTTPPort != 0 {
+		createFunctionOptions.Logger.DebugWith("Using previous deployment HTTP port ",
+			"previousHTTPPort", previousHTTPPort)
 		return previousHTTPPort, nil
 	}
 
-	// get a free local port
-	freeLocalPort, err := p.getFreeLocalPort()
-	if err != nil {
-		return -1, errors.Wrap(err, "Failed to get free local port")
-	}
-
-	p.Logger.DebugWith("Found free local port", "port", freeLocalPort)
-
-	return freeLocalPort, nil
+	return 0, nil
```

Comment:
you need -1 in this case, return this here `dockerclient.RunOptionsNoPort`
This function needs to return the correct external HTTP port to use, it basically just abstracted the call to `createFunctionOptions.FunctionConfig.Spec.GetHTTPPort()` and finding a free port (which you swapped with -1 and moved out)
If it doesn't at least do those 2 things it can be disposed of

Reply:
correct!